WordPress/WordPressShareExtension/ShareTagsPickerViewController.swift

```diff
@@ -0,0 +1,467 @@
+import Foundation
+import CocoaLumberjack
+import WordPressShared
+
+class ShareTagsPickerViewController: UIViewController {
+
+    // MARK: - Public Properties
+
+    @objc var onValueChanged: ((String) -> Void)?
+
+    // MARK: - Private Properties
+
+    /// Tags orginally passed into init()
+    ///
+    fileprivate let originalTags: [String]
+
+    /// SiteID to fetch tags for
+    ///
+    fileprivate let siteID: Int
+
+    /// Apply Bar Button
+    ///
+    fileprivate lazy var applyButton: UIBarButtonItem = {
+        let applyTitle = NSLocalizedString("Apply", comment: "Apply action on the app extension tags picker screen. Saves the selected tags for the post.")
+        let button = UIBarButtonItem(title: applyTitle, style: .plain, target: self, action: #selector(applyWasPressed))
+        button.accessibilityIdentifier = "Apply Button"
+        return button
+    }()
+
+    /// Cancel Bar Button
+    ///
+    fileprivate lazy var cancelButton: UIBarButtonItem = {
+        let cancelTitle = NSLocalizedString("Cancel", comment: "Cancel action on the app extension tags picker screen.")
+        let button = UIBarButtonItem(title: cancelTitle, style: .plain, target: self, action: #selector(cancelWasPressed))
+        button.accessibilityIdentifier = "Cancel Button"
+        return button
+    }()
+
+    @objc fileprivate let keyboardObserver = TableViewKeyboardObserver()
+    fileprivate let textView = UITextView()
+    fileprivate let textViewContainer = UIView()
+    fileprivate let tableView = UITableView(frame: .zero, style: .grouped)
+    fileprivate var dataSource: PostTagPickerDataSource = LoadingDataSource() {
+        didSet {
+            tableView.dataSource = dataSource
+            reloadTableData()
+        }
+    }
+
+    // MARK: - Initializers
+
+    init(siteID: Int, tags: String?) {
+        self.originalTags = tags?.arrayOfTags() ?? [String()]
```

Comment:
Same question above, just checking we want this to be an array with an empty string in, instead of an empty array?